include/zephyr/net/buf.h

```diff
@@ -1467,6 +1467,17 @@ static inline void * __must_check net_buf_user_data(const struct net_buf *buf)
 	return (void *)buf->user_data;
 }
 
+/**
+ * @brief Copy user data from one to another buffer.
+ *
+ * @param dst A valid pointer to a buffer gettings its user data overwritten.
+ * @param src A valid pointer to a buffer gettings its user data copied. User data size must be
+ *            equal to or exceed @a dst.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+int net_buf_user_data_clone(struct net_buf *dst, const struct net_buf *src);
```

Comment:
Wouldn't `net_buf_user_data_copy()` be a more accurate name? That's the verb you use even in the documentation, and ultimately all the implementation does is a `memcpy()`. We have a special verb for `net_buf_clone()` since it's not a pure copy, but here that's what it is in practice.

Reply:
Done as suggested